contract/views/account_analytic_account_view.xml

```diff
@@ -115,12 +115,11 @@
         <field name="model">account.analytic.account</field>
         <field name="inherit_id" ref="analytic.view_account_analytic_account_search"/>
         <field name="arch" type="xml">
+            <field name="partner_id" position="after">
+                <field name="partner_id" filter_domain="[('partner_id', 'child_of', self)]"
```

Comment:
Same note regarding the array of ids: `[self]`

Reply:
Actually wait. Shouldn't this be `[id]`?

Reply:
This self is the searchbar text